docs/dev-guide/src/config/providing.md

```diff
@@ -10,3 +10,31 @@ Flags can be set in one of four ways, in increasing order of priority:
 3. Provided individually as environment variables with the prefix `PRUSTI_` (for example, `PRUSTI_ASSERT_TIMEOUT` for the [`ASSERT_TIMEOUT`](flags.md#assert_timeout) flag).
 
 4. Provided individually as command-line arguments to Prusti with the prefix `-P` (for example, `-Pprint_desugared_specs` for the [`PRINT_DESUGARED_SPECS`](flags.md#print_desugared_specs) flag).
+
+## Multi-crate Cargo Prusti Projects
+
+Setting flags becomes slightly more complicated when Prusti is run on multiple crates as `cargo prusti` (e.g. which `Prusti.toml` file will be used). The three possible approaches to providing flags all behave differently.
+
+### Environment Variables
+
+The environment persists throughout compilation, therefore any flags set through the `DEFAULT_PRUSTI_` or `PRUSTI_` will apply to both the root crate and all dependencies. For example setting `CHECK_OVERFLOWS=false` will disable overflow checks in all dependencies, even if they try to override this with a `Prusti.toml` file.
+
+### Prusti Config File
+
+A Prusti config file is loaded prior to running cargo from `PRUSTI_CONFIG` (see default above). This file is used to configure flags in Categories B and C only and these flags apply to the entire compilation.
+
+Then, as each individual crate is checked (from the roots of the dependency tree) a Prusti config file is loaded from the current working directory (i.e. `./Prusti.toml`). This file is used to configure flags in Category A only — it would not make sense to set e.g. [`NO_VERIFY_DEPS`](flags.md#no_verify_deps) in the `./Prusti.toml` of a dependency. Cargo sets the working directory for each crate as follows:
+
+1. If the current crate is the (final) root crate, then use the directory from which `cargo prusti` was launched. Note: this case will never apply for workspace projects.
+
+2. If the current crate is in a subdirectory of the root crate/workspace, then use the directory from which `cargo prusti` was launched. Note: this means that any `Prusti.toml` in this subdirectory will be ignored!
```

Comment:
This almost looks like a bug of Prusti. I wonder if we can move the settings to a `Cargo.toml` and then reuse the toml-discovery of cargo instead of implementing it on our own.